tests/foreman/api/test_computeresource_azurerm.py

```diff
@@ -280,8 +289,158 @@ def test_positive_azurerm_host_power_on_off(self, class_host_ft, azureclient_hos
+            interfaces_attributes=self.interfaces_attributes,
+            domain=module_domain,
+            organization=module_org,
+            operatingsystem=module_os,
+            location=module_location,
+            name=self.hostname,
+            provision_method='image',
+            image=module_azurerm_cloudimg,
+            root_pass=gen_string('alphanumeric'),
+            environment=module_puppet_environment,
+            puppet_proxy=module_smart_proxy,
+            puppet_ca_proxy=module_smart_proxy,
+        ).create()
+        yield host
+        skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
+        host.delete()
+
+    @pytest.fixture(scope='class')
+    def azureclient_host(self, azurermclient, class_host_ud):
+        """Returns the AzureRM Client Host object to perform the assertions"""
+
+        return azurermclient.get_vm(name=class_host_ud.name.split('.')[0])
+
+    @upgrade
+    @tier3
+    def test_positive_azurerm_ud_host_provisioned(self, class_host_ud, azureclient_host):
+        """Host can be provisioned on AzureRm with userdata image/template
+
+        :id: df496d7c-3443-4afe-b807-5bbfc90e866e
+
+        :CaseLevel: Component
+
+        ::CaseImportance: Critical
+
+        :steps:
+            1. Create a AzureRM Compute Resource and provision host.
+
+        :expectedresults:
+            1. The host should be provisioned on AzureRM
+            2. The host name should be the same as given in data to provision the host
+            3. The host should show "Pending installation" status for provisioned host(
+               as Satellite and VM are in diff networks hence status shows as "Pending
+               installation")
+            4. The provisioned host should be assigned with external IP
+            5. The host Name and Platform should be same on Azure Cloud as provided during
+               provisioned.
+        """
+
+        assert class_host_ud.name == self.fullhostname
+        assert class_host_ud.build_status_label == "Pending installation"
+        assert class_host_ud.ip == azureclient_host.ip
+
+        # Azure cloud
+        assert self.hostname.lower() == azureclient_host.name
+        assert self.vm_size == azureclient_host.type
+
+    @upgrade
+    @tier3
+    def test_positive_host_disassociate_associate(self, class_host_ud, module_azurerm_cr):
+        """Host can be Disassociate and Associate
+
+        :id: 9514f15c-64b4-48ef-9707-fd4d39adc57d
+
+        :steps:
+            1. Disassociate a provision host and Associate it.
+
+        :expectedresults:
+            1. The host should be Disassociate
+            2. The host should be Associate
+
+        """
+
+        # Disassociate
+        disasso = class_host_ud.disassociate()
+        assert not disasso['compute_resource_name']
+
+        # Associate
+        asso = module_azurerm_cr.associate()
+        assert len(asso['results']) > 0
```

Comment:
Lets not think it will always be 1 (by saying > 0). Even though it becomes 2 its a failure. So lets do:
```suggestion
        assert len(asso['results']) == 1
```

Reply:
I did bit research on that part and because of shared org at module level or sometimes results is contains the multiples VMs details in it hence I am keeping ir in that way.